src/os/data/basedescriptor.js

```diff
@@ -698,6 +699,40 @@ os.data.BaseDescriptor.prototype.restore = function(from) {
 };
 
 
+/**
+ * @inheritDoc
+ */
+os.data.BaseDescriptor.prototype.getDescriptionMessage = function() {
```

Comment:
I think `getHTMLDescription` would be a little more fitting.

The original code from the layer menu also appears to be mostly copy/pasted from `os.ui.data.AddDataCtrl.prototype.getInfo`, which could also benefit from getting this directly from the descriptor. I updated that function recently to omit the Provider/Type lines if they're undefined/empty, so those changes should be brought in here. This does add Layer Name from the Add Data description, but I'm fine with that being in the Add Data window as well.